samples/csharp/getting-started/Ranking_PersonalizedSort/README.md

```diff
@@ -0,0 +1,215 @@
+* Information on similar competitor hotel offerings.
+
+## ML Task - Ranking
+As previously mentioned, this sample uses the LightGbm Lambdarank algorithm which is applied using a supervised learning technique known as "Learning to Rank".  This technique requires that train/test datasets contain groups of data instances that are labeled with their ideal ranking value.  The label is a numerical\ordinal value, such as {4, 3, 2, 1, 0} or a text value {"Perfect", "Excellent", "Good", "Fair", or "Bad"}.  The process for labeling these data instances with their ideal ranking value can be done manually by subject matter experts.  Or, the labels can be determined using other metrics, such as the number of clicks on a given search result. This sample uses the latter approach.
+
+Once the train/test datasets are labeled with ideal ranking values, the model (e.g. ranker) can then be trained and tested using this data.  Through the model training process, the ranker learns how to score each data instance within a group based on their label value.  The resulting score of an individual data instance by itself isn't important -- instead, the scores should be compared against one another to determine the relative ordering of a group's data instances.  The higher the score a data instance has, the more relevant and more highly ranked it is within its group.      
+
+## Solution
+The sample performs the following high-level steps to rank Expedia hotel search results:
+1. Each hotel search result is **labeled** with its ideal ranking value.
+2. Once the dataset is labeled, the data is **split** into training and testing datasets.  
+3. The model is **trained** using the train dataset using LightGbm Lambdarank algorithm.  
+4. The model is **tested** using the test dataset.  This results in a **prediction** that includes a **score** for each hotel instance.  The score is used to determine the ranking relative to other hotels within the same query (e.g. group).  The predictions are then **evaluated** by examining metrics; specifically the [Discounted Cumulative Gain](https://en.wikipedia.org/wiki/Discounted_cumulative_gain).
+5. The final step is to **consume** the model to perform ranking predictions for new incoming hotel searches.
+
+### 1. Label Data
+To label the data with ideal ranking values, the sample follows [Expedia's evaluation guidance](https://www.kaggle.com/c/expedia-personalized-sort/overview/evaluation):
+
+* 0 - The user neither clicked on this hotel nor purchased\booked a room at this hotel.
+* 1 - The user clicked through to see more information on this hotel.
+* 2 - The user purchased\booked a room at this hotel.
+
+Expedia's dataset includes both **Click_Bool** and **Booking_Bool** columns that indicate whether the user has clicked or purchased/booked a hotel.  Applying the above guidelines to these columns, we create a new **Label** column that contains values {0, 1, 2} for each hotel search result.
```

Comment:
May want to mention the the relevance gains {0, 1, 5} that these {0, 1, 2} relevance scores map to.